plugins/quickinspector/quickinspector.cpp

```diff
@@ -561,47 +574,71 @@ void QuickInspector::sgNodeDeleted(QSGNode *node)
   }
 }
 
-void QuickInspector::pickItemAt(const QPoint& pos)
+void QuickInspector::requestElementsAt(const QPoint &pos)
 {
   if (!m_window)
     return;
-  QQuickItem *item = recursiveChiltAt(m_window->contentItem(), pos);
+
+  int bestCandidate;
+  const ObjectIds objects = recursiveItemsAt(m_window->contentItem(), pos, bestCandidate);
+  if (!objects.isEmpty())
+    emit m_remoteView->elementsAtReceived(objects, bestCandidate);
+}
+
+void QuickInspector::pickElementId(const GammaRay::ObjectId &id)
+{
+  QQuickItem *item = id.asQObjectType<QQuickItem *>();
   if (item)
     m_probe->selectObject(item);
 }
 
-QQuickItem *QuickInspector::recursiveChiltAt(QQuickItem *parent, const QPointF &pos) const
+ObjectIds QuickInspector::recursiveItemsAt(QQuickItem *parent, const QPointF &pos, int &bestCandidate) const
 {
   Q_ASSERT(parent);
-  QQuickItem *child = Q_NULLPTR;
+  ObjectIds objects;
+
+  bestCandidate = -1;
 
-  // almost like QQItem::childAt, but with some extra filtering for better matching what you can see on screen
   const auto childItems = parent->childItems();
   for (int i = childItems.size() - 1; i >= 0; --i) { // backwards to match z order
-      auto c = childItems.at(i);
-      const QPointF p = parent->mapToItem(c, pos);
-      if (c->isVisible() && p.x() >= 0 && c->width() >= p.x() && p.y() >= 0 && c->height() >= p.y()) {
-          child = c;
-          // empty QQItems are less interesting, so continue looking for something better, very common first hits in ListViews for example
-          if (c->metaObject() == &QQuickItem::staticMetaObject && c->childItems().isEmpty()) {
-              continue;
-          }
-          break;
+    auto c = childItems.at(i);
+    const QPointF p = parent->mapToItem(c, pos);
+    if (c->contains(p)) {
+      const bool hasSubChildren = !c->childItems().isEmpty();
+
+      if (hasSubChildren) {
+        const int count = objects.count();
+        int bc;
+        objects << recursiveItemsAt(c, p, bc);
+
+        if (bestCandidate == -1 && bc != -1) {
+          bestCandidate = count + bc;
+        }
+      }
+      else {
+        if (bestCandidate == -1 && qGoodCandidateItem(c)) {
+          bestCandidate = objects.count();
+        }
+
+        objects << ObjectId(c);
       }
+    }
   }
 
-  if (child) {
-    return recursiveChiltAt(child, parent->mapToItem(child, pos));
+  objects << ObjectId(parent);
+
+  if (bestCandidate == -1 && qGoodCandidateItem(parent)) {
+    bestCandidate = objects.count();
```

Comment:
In the loop you want objects.count() as you add to objects after updating bestCandidate, but in this case the objects list is fully setup already. So you are off by one.